packages/maskbook/src/components/InjectedComponents/EnhancedProfile.tsx

```diff
@@ -0,0 +1,203 @@
+        const line = v.querySelector('div > div') as HTMLDivElement
+        line.style.display = ''
+    })
+}
+
+function getStyle() {
+    const eleTab = searchProfileTabSelector().evaluate()?.querySelector('div') as Element
+    const style = eleTab ? window.getComputedStyle(eleTab) : EMPTY_STYLE
+    const eleForegroundColorStyle = searchForegroundColorSelector().evaluate()
+    const foregroundColorStyle = eleForegroundColorStyle
+        ? window.getComputedStyle(eleForegroundColorStyle)
+        : EMPTY_STYLE
+
+    return {
+        color: style.color,
+        font: style.font,
+        fontSize: style.fontSize,
+        padding: style.paddingBottom,
+        height: style.height,
+        activeColor: foregroundColorStyle.color,
+    } as StyleProps
+}
+
+function useLocationChange(handler: () => void) {
+    const onLocationChange = useCallback(() => {
+        handler()
+    }, [handler])
+
+    useEffect(() => {
+        window.addEventListener('locationchange', onLocationChange)
+        return () => window.removeEventListener('locationchange', onLocationChange)
+    }, [onLocationChange])
+}
+
+export function EnhancedProfileTab(props: EnhancedProfileTabProps) {
+    const style = getStyle()
+    const { classes } = useEnhancedProfileStyles(style)
+    const [active, setActive] = useState(false)
+
+    const onClose = () => {
+        setActive(false)
+        MaskMessage.events.profileNFTsPageUpdated.sendToLocal({ show: false })
+        resetStatus()
+    }
+    const onOpen = () => {
+        MaskMessage.events.profileNFTsPageUpdated.sendToLocal({ show: true })
+        setActive(true)
+    }
+
+    useEffect(() => {
+        return MaskMessage.events.profileNFTsTabUpdated.on(onClose)
+    }, [])
+
+    const onClick = useCallback(() => {
+        onOpen()
+        clearStatus()
+    }, [clearStatus, onOpen])
+
+    return (
+        <div key="nfts" className={classes.tab}>
+            <div className={classNames(classes.button, active ? classes.hot : '')} onClick={onClick}>
+                NFTs
+                {active ? <div className={classes.active} /> : null}
+            </div>
+        </div>
+    )
+}
+const useEnhancedProfileaPageStyles = makeStyles<StyleProps>()((theme, props) => ({
+    empty: {
+        paddingTop: 36,
+        paddingBottom: 36,
+        '& > p': {
+            fontSize: 18,
+            fontFamily: 'inherit',
+            fontWeight: 700,
+            color: getMaskColor(theme).textPrimary,
+        },
+    },
+    button: {
+        backgroundColor: props.activeColor,
+        color: 'white',
+        marginTop: 18,
+        '&:hover': {
+            backgroundColor: props.activeColor,
+        },
+    },
+}))
+
+export function EnhancedProfileaPage() {
+    const [show, setShow] = useState(false)
+    const style = getStyle()
+    const { classes } = useEnhancedProfileaPageStyles(style)
+
+    useLocationChange(() => {
+        MaskMessage.events.profileNFTsTabUpdated.sendToLocal('reset')
```

Comment:
Should filter dialog URLs out. It's better to implement these logics in a  `<EnhancedProfileAtTwttier />` component.

Learn more: packages/maskbook/src/social-network-adaptor/twitter.com/collecting/getSearchedKeyword.ts